library/Makefile

```diff
@@ -21,6 +21,8 @@ endif
 # if were running on Windows build for Windows
 ifdef WINDOWS
 WINDOWS_BUILD=1
+else ifeq ($(shell uname -s),Darwin)
+APPLE_BUILD=1
```

Comment:
Should we use `?=`? Our Makefiles may not support cross-compiling, but if they do, then we may want to allow the user to force APPLE_BUILD to 0 even when their host OS is macOS. There may also be other reasons to use `?=` besides the cross compiling case.